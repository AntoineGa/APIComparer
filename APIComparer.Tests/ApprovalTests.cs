using System;
using System.IO;
using APIComparer.Outputters;
using ApprovalTests;
using NuGet;
using NUnit.Framework;

namespace APIComparer.Tests
{
    [TestFixture]
    public class ApprovalTests
    {
        [Test]
        public void ApproveMarkdown()
        {
            var packages = new[] { "4.6.3", "5.0.0-beta0002" };

            var repo = new AggregateRepository(new[] {
                PackageRepositoryFactory.Default.CreateRepository("https://www.nuget.org/api/v2"),
                PackageRepositoryFactory.Default.CreateRepository(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "NuGet", "Cache"))
            });

            var packageManager = new PackageManager(repo, "packages");

            foreach (var v in packages)
            {
                if (!Directory.Exists(Path.Combine("packages", "NServiceBus." + v)))
                    packageManager.InstallPackage("NServiceBus", SemanticVersion.Parse(v));
            }

            var file1 = Path.Combine("packages", "NServiceBus." + packages[0], "lib", "net40", "NServiceBus.Core.dll");
            var file2 = Path.Combine("packages", "NServiceBus." + packages[1], "lib", "net45", "NServiceBus.Core.dll");

            var engine = new ComparerEngine();
            engine.Filter = new NServiceBusAPIFilter();

            var diff = engine.CreateDiff(file1, file2);

            var formatter = new APIUpgradeToMarkdownFormatter("Result.md");

            formatter.WriteOut(diff);

            Approvals.Verify(File.ReadAllText("Result.md"));
        }
    }
}