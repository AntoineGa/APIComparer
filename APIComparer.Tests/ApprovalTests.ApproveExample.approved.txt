Public -> [MISSING]

Example.MissingNextVersionClass
[MISSING]

Public -> Internal

Example.InternalNextVersionClass
Example.InternalNextVersionClass

Members

- Example.MemberInternalNextVersion
  - 
[Public] string StringField
[Assembly] string StringField
  - 
[CompilerControlled, Public, HideBySig] void Method()
[CompilerControlled, Assembly, HideBySig] void Method()
  - 
[None][CompilerControlled, Public, HideBySig, SpecialName][CompilerControlled, Public, HideBySig, SpecialName] string StringProperty { get; set; }
[None][CompilerControlled, Assembly, HideBySig, SpecialName][CompilerControlled, Assembly, HideBySig, SpecialName] string StringProperty { get; set; }

- Example.MemberMissingNextVersion
  - 
[Public] string StringField
[MISSING]
  - 
[CompilerControlled, Public, HideBySig] void Method()
[MISSING]
  - 
[None][CompilerControlled, Public, HideBySig, SpecialName][CompilerControlled, Public, HideBySig, SpecialName] string StringProperty { get; set; }
[MISSING]

